completions/make.lua

```diff
@@ -0,0 +1,145 @@
+local clink_version = require('clink_version')
+if not clink_version.supports_argmatcher_delayinit then
+    print("make.lua argmatcher requires a newer version of Clink; please upgrade.")
+    return
+end
+
+-- Table of special targets to always ignore.
+local special_targets = {
+    ['.PHONY'] = true,
+    ['.SUFFIXES'] = true,
+    ['.DEFAULT'] = true,
+    ['.PRECIOUS'] = true,
+    ['.INTERMEDIATE'] = true,
+    ['.SECONDARY'] = true,
+    ['.SECONDEXPANSION'] = true,
+    ['.DELETE_ON_ERROR'] = true,
+    ['.IGNORE'] = true,
+    ['.LOW_RESOLUTION_TIME'] = true,
+    ['.SILENT'] = true,
+    ['.EXPORT_ALL_VARIABLES'] = true,
+    ['.NOTPARALLEL'] = true,
+    ['.ONESHELL'] = true,
+    ['.POSIX'] = true,
+    ['.NOEXPORT'] = true,
+    ['.MAKE'] = true,
+}
+
+local function extract_target(line, last_line, targets, allow_pathlike)
+    -- Ignore comment lines.
+    if line:find('#', 1, true) then
+        return
+    end
+
+    -- Ignore when not a target (is this only for GNU make?).
+    if last_line:find('# Not a target') then
+        return
+    end
+
+    -- Extract possible target.
+    local p = (
+        line:match('^([^%s]+):$') or -- When target has no deps.
+        line:match('^([^%s]+): '))   -- When target has deps.
+    if not p then
+        return
+    end
+
+    -- Ignore targets with pattern
+    if p:find('%%') then
+        return
+    end
+
+    -- Ignore special targets.
+    if special_targets[p] then
+        return
+    end
+
+    -- Maybe ignore path-like targets.
+    local mt
+    if allow_pathlike then
+        if not p:find('[/\\]') then
+            mt = 'alias'
+        end
+    else
+        if p:find('[/\\]') then
+            return
+        end
+    end
+
+    -- Add target.
+    table.insert(targets, { match = p, type = mt })
+end
+
+-- Sort comparator to sort pathlike targets last.
+local function comp_target_sort(a, b)
+    local a_alias = (a.type == 'alias')
+    local b_alias = (b.type == 'alias')
+    if a_alias ~= b_alias then
+        return a_alias
+    else
+        return string.comparematches(a.match, b.match)
+    end
+end
+
+local function get_targets(word, word_index, line_state, builder, user_data) -- luacheck: no unused
```

Comment:
The reason I used `_word` and `_word_index` in the nmake.lua script is because I observed that in your original make.lua script unused param names were prefixed with `_`.  I was trying to accept that convention, though I don't prefer it.

Should I abandon that convention in both the make.lua and nmake.lua scripts?

Reply:
This is my first time working with lua, so I was not sure what the general convention is and don't have any preference.
I think we can abandon _ convention, the existing files don't use that (which I should have consulted in the first place😅)